### PR TITLE
feat: fix bugs and add unittests

### PR DESCRIPTION
**Pull request by Zxilly**
fix lots of bugs and add unit tests.
I can not fix those code with filtered interface, I don't think the existing code works.
As for now, I commented out those code.

Signed-off-by: Zxilly <zhouxinyu1001@gmail.com>

### CODE REVIEW

**Comment by Zxilly** [comment]
@hsluoyz @nodece plz review.
@smrpn Could you please provide some help on filtered interfaces?

**Comment by Zxilly** [comment]
This PR is part of #3

File: package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      "type": "Apache-2.0",
+      "url": "https://www.apache.org/licenses/LICENSE-2.0"
+    }
+  ],
+  "dependencies": {
+    "@types/jest": "^26.0.23",
+    "@types/node": "^15.12.5",
+    "casbin": "^5.9.0",
+    "jest": "^27.0.5",
+    "redis": "^3.1.1"
```
Comment (nodece) [inline package.json:27]: move `casbin` and `jest` to `devDependencies`, and add `casbin` as `peerDependencies`.

File: src/adapter.ts [hunk 2]
```diff
@@ -17,14 +16,17 @@ class Line {
     v5: string = ''
 }
 
-export class NodeRedisAdapter implements FilteredAdapter {
+// noinspection FallThroughInSwitchStatementJS
+export class NodeRedisAdapter implements Adapter {
```
Comment (nodece) [inline src/adapter.ts]: `NodeRedisAdapter` should implements the `FilteredAdapter` interface.

**Comment by Zxilly** [comment]
Current not support multipie role manager. I'm trying to find a elegant way to implement it.

**Review by nodece — APPROVED** [review]

**Comment by Zxilly** [comment]
@hsluoyz plz review

**Pull request merged**